src/src/com/microsoft/aad/adal/TokenCacheKey.java

```diff
@@ -14,317 +16,309 @@
-						.getUniqueId(), item.getUserIdentifier()
-						.getDisplayableId());
-	}
-
-	/**
-	 * Store multi resource refresh tokens with different key. Key will not
-	 * include resource and set flag to y.
-	 * 
-	 * @param item
-	 *            AuthenticationRequest item
-	 * @param cacheUserId
-	 *            UserId in the cache
-	 * @return CacheKey to save token
-	 */
-	public static TokenCacheKey createMultiResourceRefreshTokenKey(
-			AuthenticationRequest item, String cacheUniqueID, String cacheDispId) {
-		if (item == null) {
-			throw new IllegalArgumentException("AuthenticationRequest");
-		}
-
-		return createCacheKey(item.getAuthority(), item.getScope(), item.getPolicy(),
-				item.getClientId(), true, cacheUniqueID, cacheDispId);
-	}
-
-	public static TokenCacheKey createCacheKey(AuthenticationRequest request,
-			AuthenticationResult result) {
-		String uniqueId = "";
-		String displayableId = "";
-
-		if (result.getUserInfo() != null) {
-			uniqueId = result.getUserInfo().getUniqueId();
-			Logger.v(TAG, "Create key with uniqueid:" + uniqueId);
-			displayableId = result.getUserInfo().getDisplayableId();
-		}
-
-		return createCacheKey(request.getAuthority(), request.getScope(), request.getPolicy(),
-				request.getClientId(), result.getIsMultiResourceRefreshToken(),
-				uniqueId, displayableId);
-	}
-
-	/**
-	 * Gets Authority.
-	 * 
-	 * @return Authority
-	 */
-	public String getAuthority() {
-		return mAuthority;
-	}
-
-	/**
-	 * Gets Scope.
-	 * 
-	 * @return Scope
-	 */
-	public String[] getScope() {
-		return mScope;
-	}
-
-	/**
-	 * Gets ClientId.
-	 * 
-	 * @return ClientId
-	 */
-	public String getClientId() {
-		return mClientId;
-	}
-
-	/**
-	 * Gets UniqueId.
-	 * 
-	 * @return UniqueId
-	 */
-	public String getUniqueId() {
-		return mUniqueId;
-	}
-
-	/**
-	 * Gets DisplayableId.
-	 * 
-	 * @return DisplayableId
-	 */
-	public String getDisplayableId() {
-		return mDisplayableId;
-	}
-
-	/**
-	 * Gets status for multi resource refresh token.
-	 * 
-	 * @return status for multi resource refresh token
-	 */
-	public boolean getIsMultipleResourceRefreshToken() {
-		return mIsMultipleResourceRefreshToken;
-	}
-
-	public void setIsMultipleResourceRefreshToken(boolean mrrt) {
-		this.mIsMultipleResourceRefreshToken = mrrt;
-	}
+    /**
+     * Serial version id.
+     */
+    private static final long serialVersionUID = 8067972995583126404L;
+
+    private static final String TAG = "TokenCacheKey";
+
+    private String mAuthority = "";
+
+    private String[] mScope;
+
+    private String mClientId = "";
+
+    private String mUniqueId = "";
+
+    private String mDisplayableId = "";
+
+    private boolean mIsMultipleResourceRefreshToken;
+
+    private String mPolicy = "";
+
+    private TokenCacheKey() {
+    }
+
+    public String toJsonString() throws JSONException {
+        JSONObject obj = new JSONObject();
+        obj.put("a", mAuthority);
+        obj.put("s", StringExtensions.createStringFromArray(mScope, " "));
```

Comment:
why does it matter in toJsonString?
Check getHashCodeForScope, mScope is converted to the treeset as we discussed, and we get the hashcode using the treeset.